clients/java/src/main/java/io/zeebe/client/impl/RetriableClientFutureImpl.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.impl;
+
+import io.grpc.stub.StreamObserver;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public class RetriableClientFutureImpl<R, T> extends ZeebeClientFutureImpl<R, T> {
+
+  private final Predicate<Throwable> retryPredicate;
+  private final Consumer<StreamObserver<T>> retryAction;
+
+  public RetriableClientFutureImpl(
+      Predicate<Throwable> retryPredicate, Consumer<StreamObserver<T>> retryAction) {
+    this(brokerResponse -> null, retryPredicate, retryAction);
+  }
+
+  public RetriableClientFutureImpl(
+      Function<T, R> responseMapper,
+      Predicate<Throwable> retryPredicate,
+      Consumer<StreamObserver<T>> retryAction) {
+    super(responseMapper);
+    this.retryPredicate = retryPredicate;
```

Comment:
Can we add `Objects.requireNonNull` on the predicate, since we always expect to have one?